marketplace/google-analytics-app/src/AppConfig.tsx

```diff
@@ -0,0 +1,302 @@
+import * as React from 'react';
+import {
+  Typography,
+  Heading,
+  Paragraph,
+  TextField,
+  Option,
+  TextLink,
+  Button,
+  Select,
+  FormLabel,
+  TextInput
+} from '@contentful/forma-36-react-components';
+import styles from './styles';
+import PropTypes from 'prop-types';
+import { AppConfigParams, AppConfigState } from './typings';
+import { getSidebarLocations } from './utils';
+
+export default class AppConfig extends React.Component<AppConfigParams, AppConfigState> {
+  static propTypes = {
+    sdk: PropTypes.object.isRequired
+  };
+
+  state: AppConfigState = {
+    allContentTypes: {},
+    contentTypes: {}
+  };
+
+  async componentDidMount() {
+    const { sdk } = this.props;
+    const { app } = sdk.platformAlpha;
+
+    const [{ items: spaceContentTypes }, savedParams] = await Promise.all([
+      sdk.space.getContentTypes().then(),
+      app.getParameters()
+    ]);
+
+    const sidebarLocations = await getSidebarLocations(sdk);
+    // remove content types for which the app has been removed from the sidebar
+    const contentTypes = sidebarLocations.reduce((acc, key) => {
+      const saved = savedParams.contentTypes[key];
+
+      if (key && saved) {
+        acc[key] = saved;
+      }
+
+      return acc;
+    }, {});
+
+    // add an incomplete contentType entry if there are none saved
+    if (!Object.keys(contentTypes).length) {
+      contentTypes[''] = { slugField: '', urlPrefix: '' };
+    }
+
+    // eslint-disable-next-line react/no-did-mount-set-state
+    this.setState(
+      {
+        allContentTypes: spaceContentTypes.reduce((acc, contentType) => {
+          contentType.fields = contentType.fields
+            // use only short text fields of content type
+            .filter(f => f.type === 'Symbol')
+            // sort by field name
+            .sort((a, b) => (a.name < b.name ? -1 : a.name > b.name ? 1 : 0));
+          acc[contentType.sys.id] = contentType;
+
+          return acc;
+        }, {}),
+        contentTypes,
+        clientId: savedParams.clientId || '',
+        viewId: savedParams.viewId || ''
+      },
+      () => sdk.app.setReady()
+    );
+
+    app.onConfigure(() => this.configureApp());
+  }
+
+  async configureApp() {
```

Comment:
We should add validation to this function. If you return `false` in this function, the app will cancel installation. I would check that a user has given a clientId, viewId, and picked at last one CT and slug field to actually install the app.

An example of showing an error would be:

```js
if (!clientId) {
  this.props.sdk.notifier.error('You must provide a valid clientId!');

  return false;
}
```
Notice I've also included an sdk method that will pop a red banner at the bottom of the page.